packages/0018-package-format.md

```diff
@@ -0,0 +1,163 @@
+
+
+#### Uniform install experience
+
+This ties into the above, but we want to ensure that the install experience is the same no matter where you install your package. Whether you drag and drop a .nupkg file into the backoffice, do it through Visual Studio or select a package in the backoffice it should feel and do the same.
+
+We want to make sure that things like the dependency management works in each case as it could otherwise open up for confusion among package consumers.
+
+#### Opt-in to changes
+
+This was mentioned briefly in the package migrations section above, but one thing we want to accomplish with this new package format is to make it more user-friendly. Currently, when you install a package there is no transparency and you have no control over what happens. Installing a package can add files, schema, content, potentially unwanted or malicious things, deletions, not handle personal data properly, etc.
+
+In an ideal world, you would install a package and get full control and insight into what it does, but that would also cause a lot of extra bloat. One thing we want to accomplish right now is to at least allow admin users to opt into when package migrations run.
+
+If the package you install or update has one or more package migrations that haven't been run you will need to authorize them with an admin account. This accomplishes two things - first is that you would have more control over things being run on the database (file changes are a bit easier to manage using version/source control such as Git).
+
+We want to show some sort of summary of what the migrations will do and present them to the user before they sign off on it.
+
+Opting in for migrations will also allow the user to plan when these things run. Can help with slow startup times due to migrations automatically running on startup.
+
+#### Warn about what will be removed on uninstall
+
+One thing that we want to see improved is the process of uninstalling a package. Currently, it can mess up your site if you uninstall a package you have used for content. Right now when you install a package it saves a list of all files and schema and content that it installed initially, and when you then uninstall it removes it all.
+
+So if you start with a starter kit and then build upon document types to make a site, and then remove the package all content built on the starter kit document types would disappear.
+
+We want to make it safer and more transparent what happens when you uninstall a package, we have several ideas but it became a huge project on its own. So, for now, the main idea is to show the user a list of things that will be removed when they try to uninstall. See more ideas under the Out of Scope section below.
+
+### Other
+
+#### NuGet feed on Our
```

Comment:
I don't think so the creation of Umbraco Our Nuget feed is a good idea, from the perspective of new developer it will introduce additional steps to start working on Umbraco development for example in visual studio. I think Our should be an aggregation of NuGet packages based on reverse dependency or based on adding packages to Our with just NuGet link :)